libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -174,12 +167,40 @@ private static string GetBlobName(string key)
                         AccessCondition.GenerateIfMatchCondition(calculatedETag),
                         new BlobRequestOptions(),
                         new OperationContext()))
-                    using (var streamWriter = new StreamWriter(blobStream))
-                    using (var jsonWriter = new JsonTextWriter(streamWriter))
+                    using (var jsonWriter = new JsonTextWriter(new StreamWriter(blobStream)))
                     {
                         JsonSerializer.Serialize(jsonWriter, newValue);
                     }
                 }));
         }
+
+        private static string GetBlobName(string key)
+        {
+            if (string.IsNullOrEmpty(key)) throw new ArgumentNullException(nameof(key));
+
+            var blobName = HttpUtility.UrlEncode(key);
+            NameValidator.ValidateBlobName(blobName);
+            return blobName;
+        }
+
+        private ValueTask<CloudBlobContainer> GetBlobContainer()

```

Comment:
>ValueTask [](start = 16, length = 9)

This is cool, I hadn't heard of this. I will have to start using this.

Reply:
Yeah, used in the right situations (and correctly) there's a potential to save a lot of allocations by using `ValueTask<T>`. Probably one of the best additions to .NET for framework writers in a while. 🎉